x-pack/legacy/plugins/maps/public/components/map/feature_tooltip.js

```diff
@@ -7,191 +7,281 @@
-      return (
-        <tr key={label}>
-          <td className="eui-textOverflowWrap mapFeatureTooltip__propertyLabel">
-            {label}
-          </td>
-          <td
-            className="eui-textOverflowWrap mapFeatureTooltip__propertyValue"
-            /*
-             * Justification for dangerouslySetInnerHTML:
-             * Property value contains value generated by Field formatter
-             * Since these formatters produce raw HTML, this component needs to be able to render them as-is, relying
-             * on the field formatter to only produce safe HTML.
-             */
-            dangerouslySetInnerHTML={{ __html: tooltipProperty.getHtmlDisplayValue() }} //eslint-disable-line react/no-danger
-          />
-          {this._renderFilterCell(tooltipProperty)}
-        </tr>
-      );
-    });
+  _renderHeader() {
+
+    if (!this.props.isLocked) {
+      return null;
+    }
 
+    const divider = (this.state.uniqueLayers && this.state.uniqueLayers.length > 1) ?
+      <EuiHorizontalRule margin="xs"/> : null;
     return (
-      <table className="mapFeatureTooltip_table">
-        <tbody>
-          {rows}
-        </tbody>
-      </table>
+      <Fragment>
+        <EuiFlexGroup justifyContent="spaceBetween">
+          <EuiFlexItem grow={false}>
+            {this._renderLayerFilterBox()}
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            {this._renderCloseButton()}
+          </EuiFlexItem>
+        </EuiFlexGroup>
+        {divider}
+      </Fragment>
     );
   }
 
-  _renderCloseButton() {
-    if (!this.props.showCloseButton) {
+  _renderFooter(filteredFeatures) {
+
+    if (filteredFeatures.length === 1) {
       return null;
     }
+
     return (
-      <EuiTextAlign textAlign="center">
-        <EuiButtonIcon
-          onClick={this.props.closeTooltip}
-          iconType="cross"
-          aria-label={i18n.translate('xpack.maps.tooltip.closeAriaLabel', {
-            defaultMessage: 'Close tooltip'
-          })}
-          data-test-subj="mapTooltipCloseButton"
-        />
-      </EuiTextAlign>
+      <Fragment>
+        <EuiHorizontalRule margin="xs"/>
+        {this._renderPagination(filteredFeatures)}
+      </Fragment>
     );
   }
 
-  render() {
-    if (!this.state.properties) {
-      const loadingMsg = i18n.translate('xpack.maps.tooltip.loadingMsg', {
-        defaultMessage: 'Loading'
-      });
-      return (
-        <EuiTextAlign textAlign="center">
-          <EuiLoadingSpinner size="m" />
-          {loadingMsg}
-        </EuiTextAlign>
-      );
+  _renderCloseButton() {
+    return (
+      <EuiButtonIcon
+        onClick={this._onCloseTooltip}
+        iconType="cross"
+        aria-label={i18n.translate('xpack.maps.tooltip.closeAriaLabel', {
+          defaultMessage: 'Close tooltip'
+        })}
+        data-test-subj="mapTooltipCloseButton"
+      />
+    );
+  }
+
+
+  _onPageChange = (pageNumber) => {
+    this.prevLayerId = null;
```

Comment:
this.prevLayerId and this.prevFeatureId no longer exists in FeatureTooltip and these two lines can be removed